### PR TITLE
[Datasets] [Docs] Remove "Datasource Compatibility" from Ray Data landing page

### LINKED ISSUE #33381: [Datasets] Remove "Datasource Compatibility" from Ray Data landing page
### Description

Almost half of the landing page describes I/O functions. This information is already described in the API reference and "Creating Datasets", so we should remove it.

image

### Link

https://docs.ray.io/en/latest/data/dataset.html

### PR DESCRIPTION
**Pull request by bveeramani**
## Why are these changes needed?

Almost half of the landing page describes I/O APIs. This information is already in the API reference and "Creating Datasets", so we should remove it.

## Related issue number

Closes #33381 

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by ericl** [comment]
Will defer to @clarkzinzow @c21 on this one.

File: doc/source/data/dataset.rst [hunk 1]
```diff
@@ -173,159 +173,6 @@ Advanced users can refer directly to the Ray Datasets :ref:`API reference <data-
         :text: Check Out Other Data Processing Options
         :classes: btn-outline-info btn-block
 
-
-.. _data-compatibility:
```
Comment (c21) [inline doc/source/data/dataset.rst:177]: can you check the repo and update all places refer to this link?

https://github.com/search?q=repo%3Aray-project%2Fray+data-compatibility&type=code

We can point users to https://docs.ray.io/en/master/data/api/input_output.html, instead.

**Review by angelinalg — APPROVED** [review]

**Comment by angelinalg** [comment]
Thank you for cleaning this up. I really appreciate your attention to maintenance, @bveeramani!

File: doc/source/data/dataset.rst [hunk 2]
```diff
@@ -173,159 +173,6 @@ Advanced users can refer directly to the Ray Datasets :ref:`API reference <data-
...
-
-Ray Datasets supports reading and writing many file formats.
-The following compatibility matrices will help you understand which formats are currently available.
-
-If none of these meet your needs, please reach out on `Discourse <https://discuss.ray.io/>`__ or open a feature
-request on the `Ray GitHub repo <https://github.com/ray-project/ray>`__, and check out
-our :ref:`guide for implementing a custom Datasets datasource <datasets_custom_datasource>`
-if you're interested in rolling your own integration!
-
-Supported Input Formats
```
Comment (pcmoritz) [inline doc/source/data/dataset.rst:191]: Have you considered just linking this section to https://docs.ray.io/en/master/data/api/input_output.html and keeping the above paragraph largely intact? It seems valuable to have a list of dataset formats linked in the landing page (but deduping as you are proposing is a good idea) :)

**Review by c21 — APPROVED** [review]
LGTM

**Pull request merged**